hr_attendance_overtime_hours/models/hr_attendance_overtime.py

```diff
@@ -0,0 +1,40 @@
+# Copyright 2023 Janik von Rotz <janik.vonrotz@mint-system.ch>
+# Copyright 2024 Camptocamp
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from datetime import datetime, time
+
+from odoo import api, fields, models
+
+
+class HrAttendanceOvertime(models.Model):
+    _inherit = "hr.attendance.overtime"
+
+    planned_hours = fields.Float(compute="_compute_planned_hours", store=True)
+    worked_hours = fields.Float(compute="_compute_worked_hours", store=True)
+
+    @api.depends("date")
+    def _compute_planned_hours(self):
+        for overtime in self:
+            # Get work hours from calendar
+            planned_hours = (
+                overtime.employee_id.resource_calendar_id.get_work_hours_count(
+                    datetime.combine(overtime.date, time.min),
+                    datetime.combine(overtime.date, time.max),
+                    True,
+                )
+            )
+            overtime.planned_hours = planned_hours
+
+    @api.depends("duration")
+    def _compute_worked_hours(self):
+        for overtime in self:
+            # Get sum of attendance entries
+            attendance_ids = overtime.env["hr.attendance"].search(
+                [
+                    ("employee_id", "=", overtime.employee_id.id),
+                    ("check_in", ">=", datetime.combine(overtime.date, time.min)),
+                    ("check_out", "<=", datetime.combine(overtime.date, time.max)),
```

Comment:
We should take this occasion to improve this piece of code. Now, it does not take into account the timezone. 
`check_in` should be compared to `dt_start` from below suggestion.
(you make the same for `check_out`)
```
        tz = overtime.employee_id.resource_calendar_id.tz or overtime.employee_id.tz or self.env.user.tz
        date_tz = datetime.combine(overtime.date, time.min)
        # Convert to UTC
        dt_start = fields.Datetime.from_string(
            self.env["ir.fields.converter"]
            .with_context(tz=tz)
            ._str_to_datetime(None, None, dt_start_tz.replace(tzinfo=None))[0]
        )
```

Reply:
Please test it, i just test by eyes.

Reply:
hello @nguyenminhchien, I adjusted as per your suggestion and split the functions into smaller ones for easier handling. thank for ur suggestions